code/game/objects/items/rcd/RPD.dm

```diff
@@ -609,6 +596,69 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 			else
 				return ..()
 
+/obj/item/pipe_dispenser/proc/check_can_make_pipe(atom/target_of_attack)
+	//make sure what we're clicking is valid for the current category
+	var/static/list/make_pipe_whitelist
+	if(!make_pipe_whitelist)
+		make_pipe_whitelist = typecacheof(list(/obj/structure/lattice, /obj/structure/girder, /obj/item/pipe, /obj/structure/window, /obj/structure/grille))
```

Comment:
```suggestion
	var/static/list/make_pipe_whitelist = typecacheof(list(/obj/structure/lattice, /obj/structure/girder, /obj/item/pipe, /obj/structure/window, /obj/structure/grille))
```
This should work just fine, static lists are filled when the server/world is booted and `typecacheof` doesn't really rely on anything created after that (eg. global lists).